libaleth-interpreter/VM.cpp

```diff
@@ -350,6 +350,8 @@ void VM::interpretCases()
             // 0-value selfdestructs don't have to pay this charge.
             if (m_rev >= EVMC_TANGERINE_WHISTLE)
             {
+                updateIOGas();
```

Comment:
Alternative approach:
```
updateIOGas();

if (m_rev >= EVMC_TANGERINE_WHISTLE)
{
  if (m_rev == ...)
  {
    if (!m_context-> ...)
    {
       m_runGas = VMSchedule::callNewAccount;
       updateIOGas();
    }
  }
}
```
This seems cleaner to me, because it's clearly seen that we set `m_runGas` + call `updateIOGas()` the second time to charge additional gas. Also this is more similar to another place where we charge gas in two phases - that's for CALL opcodes in `caseCallSetup` (first basic gas + charge for memory, then the gas that the caller provides to subframe)

Reply:
Makes perfect sense!